transform/common.go

```diff
@@ -24,3 +29,47 @@ func upper(name string) interface{} {
 func lower(name string) interface{} {
 	return strings.ToLower(name)
 }
+
+// truncate returns the first n characters of str. If n is greater than the
+// length of str or less than 0, return str.
+func truncate(str string, n int) string {
+	if len(str) < n || n < 0 {
+		return str
+	}
+
+	return str[0:n]
+}
+
+// FindHash returns a func to create a new hash based on the provided name.
+func FindHash(name string) func() hash.Hash {
+	switch strings.ToUpper(name) {
+	case "SHA1":
+		return sha1.New
+	}
+	return nil
+}
+
+// ComputeHash applies the hash h to the file located at path. Returns a line
+// of dashes for directories.
+func ComputeHash(info os.FileInfo, path string, h hash.Hash) (interface{}, error) {
+	if info.IsDir() {
+		return strings.Repeat("-", h.Size()*2), nil
+	}
+
+	f, err := os.Open(path)
+	if err != nil {
+		return nil, err
+	}
+	defer f.Close()
```

Comment:
it may be worth refactoring this so we don't have to open up the file here, not sure if we actually want to pass around an open file though